express-notes-api/server.js

```diff
@@ -0,0 +1,109 @@
+import express from 'express';
+import { readFile, writeFile } from 'fs/promises';
+
+const app = express();
+app.use(express.json());
+
+const dataPath = 'data.json';
+
+app.get('/api/notes', async (req, res) => {
+  try {
+    const data = await readFile(dataPath, 'utf8');
+    const notes = JSON.parse(data).notes;
+    res.status(200).json(Object.values(notes));
+  } catch (err) {
+    console.error(err);
+    res.status(500).json({ error: 'Error reading data file' });
+  }
+});
+
+app.get('/api/notes/:id', async (req, res) => {
+  const id = Number(req.params.id);
+  if (isNaN(id) || id <= 0) {
+    res.status(400).json({ error: 'id must be a positive integer' });
+  } else {
+    try {
+      const data = await readFile(dataPath, 'utf8');
+      const notes = JSON.parse(data).notes;
+      if (notes[id]) {
+        res.status(200).json(notes[id]);
+      } else {
+        res.status(404).json({ error: `cannot find note with id ${id}` });
+      }
+    } catch (err) {
+      console.error(err);
+      res.status(500).json({ error: 'Error reading data file' });
+    }
+  }
+});
+
+app.post('/api/notes', async (req, res) => {
+  const content = req.body.content;
+  if (!content) {
+    res.status(400).json({ error: 'content is a required field' });
+  } else {
+    try {
+      const data = await readFile(dataPath, 'utf8');
+      const notes = JSON.parse(data).notes;
+      const id = Math.floor(Math.random() * 100) + 1;
```

Comment:
This isn't guaranteed to make them unique. The intent was to use the `nextId` property in the data.json file. That seems to have disappeared, though. What happened?